File: 03.oop.lua
local Account = {
  balance = 0,
  add = function(acct,nr)
    acct.balance = acct.balance + nr
  end,
  sub = function(acct,nr)
    acct.balance = acct.balance - nr
  end
}

local my_account = { balance = 0 }

print(my_account.balance)

Account.add(my_account,5)

print(my_account.balance)

Account.sub(my_account,6)

print(my_account.balance)

-- my_account:add(5)

local brou_account = { balance = 0 }
setmetatable(brou_account,{__index = Account})

print(brou_account.balance)

brou_account:add(5)

print(brou_account.balance)

brou_account:sub(6)

print(brou_account.balance)

File: 03.oop.js
var Account = {
  add: function(nr) {
    this.balance = this.balance + nr
  },
  sub: function(nr) {
    this.balance = this.balance - nr
  }
}

var my_account = { balance: 0 }

my_account.__proto__ = Account

console.log(my_account.balance)

my_account.add(5)

console.log(my_account.balance)

my_account.sub(6)

console.log(my_account.balance)
